sda-commons-server-auth/src/main/java/org/sdase/commons/server/auth/service/AuthRSA256Service.java

```diff
@@ -38,28 +47,31 @@ public AuthRSA256Service(RsaPublicKeyLoader rsaPublicKeyLoader, long leeway) {
       String keyId = JWT.decode(authorizationToken).getKeyId();
       if (keyId == null) {
         // check all keys without id
-        List<RSAPublicKey> keysWithoutId = rsaPublicKeyLoader.getKeysWithoutId();
+        List<LoadedPublicKey> keysWithoutId = rsaPublicKeyLoader.getKeysWithoutId();
         if (keysWithoutId.size() > 1) {
-          log.warn("Verifying token without kid trying {} public keys", keysWithoutId.size());
+          LOG.warn("Verifying token without kid trying {} public keys", keysWithoutId.size());
         }
         Collections.reverse(keysWithoutId);
         return keysWithoutId.stream()
-            .map(k -> verifyJwtSignature(authorizationToken, k))
-            .filter(Optional::isPresent)
+            .map(k -> Pair.of(verifyJwtSignature(authorizationToken, k.getPublicKey()), k))
+            .filter(p -> p.getKey().isPresent())
+            .filter(p -> validateIssuer(p.getKey().get(), p.getValue()))
```

Comment:
You could skip this manual validation (also in the `else` branch). [`JWT.require(…)`](https://github.com/SDA-SE/sda-dropwizard-commons/pull/1030/files#diff-d7e2444a83694da1cbd5193621217ca2afee1b9091ec0b1e621a1cf35457eb4bR165) that is used in `verifyJwtSignature(…)` has a method `withIssuer(…)` that is doing exactly this validation.